### PR TITLE
azurerm_api_management: Enable Tenant Access

### PR DESCRIPTION
**Pull request by patst**
Implementation for https://github.com/terraform-providers/terraform-provider-azurerm/issues/8578 .

Management REST API can be enabled/ disabled and the primary/ secondary Keys can be used as output variables.

Feedback welcome ;-)

### CODE REVIEW

**Comment by patst** [comment]
integration test passed in my subscription

**Review by katbyte — CHANGES_REQUESTED** [review]
THanks for the pr @patst - are you sure the tests pass? they all appear to be failing for us with the tenant_access block being updated.
```
enant_access.#: "1" => "0"
```

**Comment by patst** [comment]
@katbyte thanks for the feedback. Only the "positive test" seems to work.

The problems is to separate between a disabled `tenant_access` with
```
tenant_access {
 enabled= false
}
```
and a missing `tenant_access` block. The REST API is returning for both cases the same response.

I will try to figure out a solution. I think the `flattenApiManagementTenantAccessSettings` function is the place to look for a solution.

If you have any advise that would be great as well

**Comment by patst** [comment]
I changed the code and the tests I executed passed

```
TF_ACC=1 go test ./azurerm/internal/services/apimanagement/api_management_resource_test.go -v -run=TestAccApiManagement_customProps -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccApiManagement_customProps
=== PAUSE TestAccApiManagement_customProps
=== CONT  TestAccApiManagement_customProps
--- PASS: TestAccApiManagement_customProps (2985.12s)
PASS
ok      command-line-arguments  2986.535s

TF_ACC=1 go test ./azurerm/internal/services/apimanagement/api_management_resource_test.go -v -run=TestAccApiManagement_tenantAccess -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccApiManagement_tenantAccess
=== PAUSE TestAccApiManagement_tenantAccess
=== CONT  TestAccApiManagement_tenantAccess
--- PASS: TestAccApiManagement_tenantAccess (2593.46s)
PASS

TF_ACC=1 go test ./azurerm/internal/services/apimanagement/api_management_resource_test.go -v -run=TestAccApiManagement_signInSignUpSettings -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccApiManagement_signInSignUpSettings
=== PAUSE TestAccApiManagement_signInSignUpSettings
=== CONT  TestAccApiManagement_signInSignUpSettings
--- PASS: TestAccApiManagement_signInSignUpSettings (2766.83s)
PASS
```

I guess the remaining ones will pass as well.

File: website/docs/r/api_management.html.markdown [hunk 1]
```diff
@@ -299,6 +301,12 @@ A `sign_up` block supports the following:
 
 ---
 
+A `tenant_access` block supports the following:
+
+* `enabled` - (Required) Should the access to the management api be enabled?
+
```
Comment (katbyte) [inline website/docs/r/api_management.html.markdown:307]: we are missing the rest of the blocks propeties here?

File: azurerm/internal/services/apimanagement/api_management_resource.go [hunk 2]
```diff
@@ -528,6 +528,35 @@ func resourceApiManagementService() *schema.Resource {
...
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"id": {
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/api_management_resource.go:542]: might be more clear if we add tenant here
```suggestion
						"tenant_id": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @patst - tests are looking better now, just a couple coments more inline to address and this should be good ro merge

File: azurerm/internal/services/apimanagement/api_management_resource.go (hunk 2 above)
Comment (patst) [reply azurerm/internal/services/apimanagement/api_management_resource.go]: good idea

File: website/docs/r/api_management.html.markdown (hunk 1 above)
Comment (patst) [reply website/docs/r/api_management.html.markdown:307]: The missing properties are only exported and cannot be set. I added them to the 'Attribute reference' part of the docs.

**Review by katbyte — APPROVED** [review]
Thanks @patst - LGTM

**Pull request merged**